src/Framework/AspNetCoreAnalyzers/test/Mvc/AuthorizeAttributeOverriddenTest.cs

```diff
@@ -0,0 +1,602 @@
+{
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task CustomAuthorizeCombinedWithAllowAnonymousOnAction_AllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[AllowAnonymous]
+public class MyController
+{
+    [MyAuthorize]
+    public object Get() => new();
+}
+
+public class MyAuthorizeAttribute : Attribute, IAuthorizeData, IAllowAnonymous
+{
+    public string? Policy { get; set; }
+    public string? Roles { get; set; }
+    public string? AuthenticationSchemes { get; set; }
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task AuthorizeBeforeAllowAnonymousOnAction_AllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[AllowAnonymous]
+public class MyController
+{
+    [Authorize(AuthenticationSchemes = "foo")]
+    [AllowAnonymous]
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task AuthorizeAfterAllowAnonymousOnAction_NoAttributeOnController_HasDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+public class MyController
+{
+    [AllowAnonymous]
+    [{|#0:Authorize|}]
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source,
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyController.Get").WithLocation(0)
+        );
+    }
+
+    [Fact]
+    public async Task NoAttributeOnAction_AuthorizeBeforeAllowAnonymousOnController_NoDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[Authorize(AuthenticationSchemes = "foo")]
+[AllowAnonymous]
+public class MyController
+{
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source);
+    }
+
+    [Fact]
+    public async Task NoAttributeOnAction_AuthorizeAfterAllowAnonymousOnController_HasDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+[AllowAnonymous]
+[{|#0:Authorize|}]
+public class MyController
+{
+    public object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source,
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyController").WithLocation(0)
+        );
+    }
+
+    [Fact]
+    public async Task AuthorizeOnAction_AllowAnonymousOnBaseMethod_HasDiagnostics()
+    {
+        var source = $$"""
+{{CommonPrefix}}
+public class MyControllerBase
+{
+    [AllowAnonymous]
+    public virtual object Get() => new();
+}
+
+public class MyController : MyControllerBase
+{
+    [{|#0:Authorize|}]
+    public override object Get() => new();
+}
+""";
+
+        await VerifyCS.VerifyAnalyzerAsync(source,
+            new DiagnosticResult(DiagnosticDescriptors.AuthorizeAttributeOverridden).WithArguments("MyControllerBase.Get").WithLocation(0)
+        );
+    }
```

Comment:
Do we want to also cover situations where the signature is hidden? 

```csharp
[Route("[controller]")]
public class MyBaseController : Controller
{
    public IActionResult Index()
    {
        return View();
    }
}

public class HomeController : MyBaseController
{
    public new IActionResult Index()
    {
        return View();
    }
```

Reply:
In the `new` case, the closer attributes always win, right? And that's what we think people expect. So, there should be no relevant "farther away" [AllowAnonymous] attributes to warn about.

Reply:
Sorry, my example got botched at some point. I've updated the comment to reflect that. Maybe I put the comment in the wrong place and was misinterpreted as related to overriding, but I was suggesting that we also covered the case with new, even if we "know" it works like other cases to ensure that if we find this case we don't produce the wrong result.